class.jetpack.php

```diff
@@ -5336,7 +5248,14 @@ public static function validate_sync_error_idc_option() {
 
 		// Is the site opted in and does the stored sync_error_idc option match what we now generate?
 		if ( $sync_error && self::sync_idc_optin() ) {
-			$error_diff = array_diff_assoc( $sync_error, self::get_sync_error_idc_option() );
+			if ( isset( $sync_error['home'], $sync_error['siteurl'] ) ) {
```

Comment:
Since we're changing how we generate the validated object, should we add tests for when we only store the home URL option and when we only store the siteurl option?